contrib/babelfishpg_tds/src/backend/tds/guc.c

```diff
@@ -111,6 +113,62 @@ TdsGucDefaultPacketSizeCheck(int *newvalue, void **extra, GucSource source)
 	return true;
 }
 
+static bool 
+check_version_number(char **newval, void **extra, GucSource source)
+{
+	char 		*copy_version_number = malloc(sizeof(*newval));
+	char		*token;
+	int		part = 0;
+
+    strcpy(copy_version_number,*newval);
+	if(strcasecmp(copy_version_number,"default") == 0)
+		return true;
+
+	for (token = strtok(copy_version_number, "."); token; token = strtok(NULL, "."))
+	{	
+		/* check each token contains only digits */
+		if(strspn(token, "0123456789") != strlen(token))
```

Comment:
1. Can we use `isdigit()`?
2. What if `token` is an empty string?

Reply:
Yes, we can use ``isdigit()``. And if ``token`` is an empty srting, it will throw an error using ``isdigit()``, so the error message will be ``ERROR: Please enter a valid version number``